vulkan/vulkan_proc_table.cc

```diff
@@ -0,0 +1,99 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "vulkan_proc_table.h"
+#include "lib/ftl/logging.h"
+
+#include <dlfcn.h>
+
+namespace vulkan {
+
+VulkanProcTable::VulkanProcTable() : valid_(false), handle_(nullptr) {
+  if (!OpenLibraryHandle()) {
+    return;
+  }
+
+  if (!AcquireProcs()) {
+    return;
+  }
+
+  valid_ = true;
+}
+
+VulkanProcTable::~VulkanProcTable() {
+  CloseLibraryHandle();
+}
+
+bool VulkanProcTable::IsValid() const {
+  return valid_;
+}
+
+bool VulkanProcTable::OpenLibraryHandle() {
+  dlerror();  // clear existing errors on thread.
+  handle_ = dlopen("libvulkan.so", RTLD_NOW);
+  if (handle_ == nullptr) {
+    FTL_LOG(INFO) << "Could not open the vulkan library: " << dlerror();
```

Comment:
Probably a WARNING rather than an INFO.

Will this print all the time on Android?  We might want to fail silently rather than spamming the log.